File: test-suite-python/src/main/python/helloworld/ServerInterceptorFactory.py
# tag::imports[]
from io.grpc import ServerInterceptor
from jakarta.inject import Singleton
from micronaut.context.annotation import Bean, Factory
from micronaut.grpc.server.interceptor import OrderedServerInterceptor

from .CustomInterceptor import CustomInterceptor
# end::imports[]
from micronaut.context.annotation import Requires


@Requires(property="spec.name", value="ServerInterceptorTest")
# tag::clazz[]
@Factory  # <1>
class ServerInterceptorFactory:

    @Bean  # <2>
    @Singleton
    def custom_server_interceptor(self) -> ServerInterceptor:
        return OrderedServerInterceptor(CustomInterceptor(), 10)  # <3>
# end::clazz[]

File: test-suite/src/main/java/helloworld/ServerInterceptorFactory.java
package helloworld;

// tag::imports[]
import io.grpc.ServerInterceptor;
import io.micronaut.context.annotation.Bean;
import io.micronaut.context.annotation.Factory;
import io.micronaut.grpc.server.interceptor.OrderedServerInterceptor;
import jakarta.inject.Singleton;
// end::imports[]
import io.micronaut.context.annotation.Requires;

@Requires(property = "spec.name", value = "ServerInterceptorTest")
// tag::clazz[]
@Factory // <1>
public class ServerInterceptorFactory {

    @Bean // <2>
    @Singleton
    public ServerInterceptor customServerInterceptor() {
        return new OrderedServerInterceptor(new CustomInterceptor(), 10); // <3>
    }
}
// end::clazz[]
